File: .sdk-repos/orchestration-cluster-api-csharp/examples/MappingRule.cs
// Compilable usage examples for mapping rule operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class MappingRuleExamples
{
    #region CreateMappingRule
    // <CreateMappingRule>
    public static async Task CreateMappingRuleExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.CreateMappingRuleAsync(new MappingRuleCreateRequest
        {
            ClaimName = "groups",
            ClaimValue = "engineering",
            Name = "Engineering Group Mapping",
        });

        Console.WriteLine($"Mapping rule: {result.MappingRuleId}");
    }
    // </CreateMappingRule>
    #endregion CreateMappingRule

    #region GetMappingRule

    // <GetMappingRule>
    public static async Task GetMappingRuleExample(MappingRuleId mappingRuleId)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetMappingRuleAsync(mappingRuleId);
        Console.WriteLine($"Mapping rule: {result.Name}");
    }
    // </GetMappingRule>
    #endregion GetMappingRule

    #region SearchMappingRule

    // <SearchMappingRule>
    public static async Task SearchMappingRuleExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchMappingRuleAsync(
            new MappingRuleSearchQueryRequest());

        foreach (var rule in result.Items)
        {
            Console.WriteLine($"Mapping rule: {rule.Name}");
        }
    }
    // </SearchMappingRule>
    #endregion SearchMappingRule

    #region UpdateMappingRule

    // <UpdateMappingRule>
    public static async Task UpdateMappingRuleExample(MappingRuleId mappingRuleId)
    {
        using var client = CamundaClient.Create();

        await client.UpdateMappingRuleAsync(mappingRuleId, new MappingRuleUpdateRequest
        {
            ClaimName = "groups",
            ClaimValue = "senior-engineering",
            Name = "Senior Engineering Mapping",
        });
    }
    // </UpdateMappingRule>
    #endregion UpdateMappingRule

    #region DeleteMappingRule

    // <DeleteMappingRule>
    public static async Task DeleteMappingRuleExample(MappingRuleId mappingRuleId)
    {
        using var client = CamundaClient.Create();

        await client.DeleteMappingRuleAsync(mappingRuleId);
    }
    // </DeleteMappingRule>
    #endregion DeleteMappingRule
}

File: .sdk-repos/orchestration-cluster-api-php/examples/mapping_rule.php
<?php

/**
 * Compilable usage examples for mapping-rule operations.
 */

declare(strict_types=1);

namespace Camunda\Orchestration\Examples;

use Camunda\Orchestration\CamundaClient;

// region CreateMappingRule
/**
 * Create mapping rule.
 */
function create_mapping_rule(CamundaClient $client, ?\Camunda\Orchestration\Api\Model\MappingRuleCreateRequest $mappingRuleCreateRequest = null): void
{
    $client->createMappingRule($mappingRuleCreateRequest);
}
// endregion CreateMappingRule

// region SearchMappingRule
/**
 * Search mapping rules.
 */
function search_mapping_rule(CamundaClient $client, ?\Camunda\Orchestration\Api\Model\MappingRuleSearchQueryRequest $mappingRuleSearchQueryRequest = null): void
{
    $client->searchMappingRule($mappingRuleSearchQueryRequest);
}
// endregion SearchMappingRule

// region GetMappingRule
/**
 * Get a mapping rule.
 */
function get_mapping_rule(CamundaClient $client, string $mappingRuleId): void
{
    $client->getMappingRule($mappingRuleId);
}
// endregion GetMappingRule

// region UpdateMappingRule
/**
 * Update mapping rule.
 */
function update_mapping_rule(CamundaClient $client, string $mappingRuleId, ?\Camunda\Orchestration\Api\Model\MappingRuleUpdateRequest $mappingRuleUpdateRequest = null): void
{
    $client->updateMappingRule($mappingRuleId, $mappingRuleUpdateRequest);
}
// endregion UpdateMappingRule

// region DeleteMappingRule
/**
 * Delete a mapping rule.
 */
function delete_mapping_rule(CamundaClient $client, string $mappingRuleId): void
{
    $client->deleteMappingRule($mappingRuleId);
}
// endregion DeleteMappingRule
